lib/parsers/gomod-graph-parser.ts

```diff
@@ -0,0 +1,32 @@
+import {DepGraphBuilder, DepGraph, PkgInfo} from '@snyk/dep-graph';
+
+function parseGoModGraphLine(line: string) {
+  return line
+      .trim()
+      .split(/\s/)
```

Comment:
```javascript
function parseGoModGraphLine(line: string) {
  return line.trim().split(/\s|@/);
}

// usage
const [parentName, parentVersion, childName, childVersion] = parseGoModGraphLine(line);
```

Reply:
Not sure what you mean?

Reply:
you don't need to split twice, you can split only once on two tokens

Reply:
You have to. Why?
The go.mod lines are of the shape:
* `github.com/parent/module github.com/child/module@v1.1.1`
* `github.com/parent/module@v1.6.1 github.com/child/module@v1.1.0`

For bullet 2, if we'll split by `@` at the start, we'll have
```ts
['github.com/parent/module', 'v1.6.1 github.com/child/module', 'v1.1.0']
```
Which is bad for us to interpret.